pkg/daemon/ceph/client/pool.go

```diff
@@ -232,14 +236,28 @@ func givePoolAppTag(context *clusterd.Context, namespace string, poolName string
 }
 
 func setCommonPoolProperties(context *clusterd.Context, pool cephv1.PoolSpec, namespace, poolName, appName string) error {
-	compressionModeProperty := "compression_mode"
-	if pool.CompressionMode != "" {
+	if pool.Replicated.IsTargetRatioEnabled() {
+		err := SetPoolProperty(context, namespace, poolName, targetSizeRatioProperty, strconv.FormatFloat(pool.Replicated.TargetSizeRatio, 'f', -1, 32))
+		if err != nil {
+			return errors.Wrapf(err, "failed to set property %q to pool %q to %.2f", targetSizeRatioProperty, poolName, pool.Replicated.TargetSizeRatio)
+		}
+	}
+
+	if pool.IsCompressionEnabled() {
 		err := SetPoolProperty(context, namespace, poolName, compressionModeProperty, pool.CompressionMode)
 		if err != nil {
 			return errors.Wrapf(err, "failed to set property %q on pool %q to %q", compressionModeProperty, poolName, pool.CompressionMode)
 		}
 	}
 
+	// Apply additional properties
+	for _, prop := range pool.PoolProperties {
+		err := SetPoolProperty(context, namespace, poolName, prop.Name, prop.Value)
+		if err != nil {
+			logger.Errorf("failed to set property %q to pool %q to %q. %v", prop.Name, poolName, prop.Value)
```

Comment:
We should have some visibility into the properties that succeeded or failed being applied to the pool. What if we added to the `status` section of the CR to indicate the properties that were applied? This will be a bit more complex since this same code path is used for rgw and mds pools as well, but adding the pool status to those CRs would be good as well. And this certainly could be a separate PR.

Reply:
I can try to differentiate between a failure when applying property and when a property does not exist.